### ISSUE TITLE
  {"_action":"_ask","error":"an error occurred HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x000001F52CF32190>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it')))","success":false}

### ISSUE THREAD

**Issue by xinoxme** [xtekky/chatgpt-clone#79]

**Comment by xinoxme** [comment]
![Screenshot (1145)](https://user-images.githubusercontent.com/127733344/235447523-e21ebe4f-b9c9-478d-b480-0950d07cefe8.png)

**Comment by maxens-git** [comment]
I have the same message error

**Comment by neildmd** [comment]
I have almost the identical error. 

`{"_action":"_ask","error":"an error occurred HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f943d6ae970>: Failed to establish a new connection: [Errno 111] Connection refused')))","success":false}`

The default config.json contains this section:

`"openai_api_base": "https://api.openai.com",
    "proxy": {
        "enable": true,
        "http": "127.0.0.1:7890",
        "https": "127.0.0.1:7890"
    }`
    
It looks like it's expecting a proxy to be running locally on port 7890. I tried setting the enable option to `false` but I still get the error above.

**Comment by metadreamx** [comment]
Getting the same error, have not seen any feed back on this error so far.

**Comment by jp7189** [comment]
Same here.

{"_action":"_ask","error":"an error occurred HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x0000020E2C832C50>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it')))","success":false}

**Comment by chekamarue** [comment]
same here 

{"_action":"_ask","error":"an error occurred HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x00000245CD3E3610>: Failed to establish a new connection: [WinError 10061] No connection could be made because the target machine actively refused it')))","success":false}

**Comment by chekamarue** [comment]
edit config.json and enter api key ! 
set False for proxy

**Comment by jp7189** [comment]
api key was already in there.
I set proxy to false, but that gets me: 
"I apologize, but it seems that I am having difficulty connecting to the OpenAI server. Please check your internet connection and try again later."

..and in the console:
"
list index out of range
None
<redacted IP> - - [01/May/2023 15:13:59] "POST /backend-api/v2/conversation HTTP/1.1" 200 -
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range
None
Expecting value: line 1 column 2 (char 1)
<traceback object at 0x000002077DD0D040>
list index out of range
None
"

**Comment by jp7189** [comment]
Following up... I cleared all conversations in the web gui and it's working properly now.
Thanks for the help.
The fix: set the proxy false in config.json
    "proxy": {
        "enable": false,

**Comment by MortiMer0101** [comment]
When I ask something, it never answers me. Any help?

**Comment by xinoxme** [comment]
> edit config.json and enter api key ! set False for proxy

it did not work

**Comment by jp7189** [comment]
Check out the author's other project - gpt4free - that's the reverse engineered API.  This isn't that.

**Comment by jp7189** [comment]
> > edit config.json and enter api key ! set False for proxy
> 
> it did not work

I had to restart the script, restart the browser, and clear all conversations to get it working after the changes.

**Comment by xinoxme** [comment]
> > > edit config.json and enter api key ! set False for proxy
> > 
> > 
> > it did not work
> 
> I had to restart the script, restart the browser, and clear all conversations to get it working after the changes.

did it work?

**Comment by luisf371** [comment]
Because of the way docker works, the config.json is burned INTO the image when you first build it, you can create a hard link between your local .json and the one inside the docker image, or you can re-build the image with the update config.json with the steps below.
--
1) docker-compose down
2) make changes to config.json
3) docker-compose up -d --build

-d starts it in daemon mode (background) and --build forces to rebuild the image.

**Comment by jp7189** [comment]
> > > > edit config.json and enter api key ! set False for proxy
> > > 
> > > 
> > > it did not work
> > 
> > 
> > I had to restart the script, restart the browser, and clear all conversations to get it working after the changes.
> 
> did it work?

Yes.  It worked fine... two days ago.  When I tried it today (with no changes) I don't get any response at all. No errors, it just sticks on generating forever.

**Comment by duringleaves** [comment]
See the same problem.  config.json looks fine, API key is good.  cleared conversations, restarted script.  no change.

**Comment by duringleaves** [comment]
set proxy: false and it works a little better now except the console shows a lot of "list index out of range" errors now.  gpt-35-turbo works, but gpt-4 fails

**Comment by NOONE1526** [comment]
after i changed true to false all modes stopped working

**Comment by Mario-119** [comment]
> after i changed true to false all modes stopped working

Likewise in my case, I just get no response no matter if I use gpt-3.5 or gpt-4

**Comment by SidJnsn66** [comment]
Luisf371's comment on May 4 corrected the problem for me, after changing the proxy term to false in config.json.  Thank you, I would have never figured that out!

**Comment by fsantamaria1** [comment]
I tried all the solutions above. The GUI does not respond after asking a questions; however, I see this in the console:

"POST /backend-api/v2/conversation HTTP/1.1" 200 -
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range
None
list index out of range

**Comment by fsantamaria1** [comment]
If you have set the `proxy` to false and added the `API key` and it still doesn't work, then it could be an issue related to your OpenAI account since they only let you use the API for free for the `first three months` after account creation. Unfortunately, the only solution to that issue is adding a payment method [here](https://platform.openai.com/account/billing/overview). Keep in mind that you can set usage limits [here](https://platform.openai.com/account/billing/limits).

**Comment by GuoWei96** [comment]
Thank you for your open-source project. I have identified a bug
When the input content is too much, it will exceed the range of the input box
